workflow-templates/im-test-playwright-manual.yml

```
@@ -0,0 +1,224 @@
+#    With the workflow the user specifies when they kick it off manually.
+#
+# Frequency:
+#    - This workflow is intended to be reusable, and used once per repository.
+#
+# Projects to use this Template with:
+#    - Testing projects that use Playwright.
+# # TODO: Prerequisites:
+#    - Add @estruyf/github-actions-reporter reporter to your playwright config when running in CI.  npm install --dev @estruyf/github-actions-reporter.
+#       - playwright.config.ts Example:  
+#         reporter: process.env.CI ? [
+#             ['list'],
+#             ['html'],
+#             ['@estruyf/github-actions-reporter'] 
+#           ] : [['list'],['html']], 
+
+name: 🧪 Run Playwright Tests
+run-name: 🧪 Run Playwright Tests on ${{ inputs.environment == 0 && 'dev' || inputs.environment }} with ${{ inputs.browser-project == 0 && 'chromium' || inputs.browser-project }} browser
+
+concurrency:
+  group: playwright-${{ github.event_name == 'workflow_dispatch' && github.run_id || github.ref  }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch-tag-sha:
+        description: Branch/Tag/SHA; The branch, tag or sha of the selenium tests should be run against.
+        required: false
+      environment:
+        description: The environment to test against
+        required: true
+        type: choice
+        options:
+          - dev
+          - qa
+          - stage
+          # - prod # Uncomment this line if you want to test against prod
+      browser-project:
+        description: Browser - what browser project to use. # Browser projects defined in the playwright.config.ts file.
+        type: choice
+        options:
+          - chromium
+          - firefox
+          - webkit
+          - all
+      config-file:
+        description: Config File; The config file to use for the tests.
+        required: false
+        type: choice
+        default: playwright.config.ts
+        options:
+          - playwright.config.ts
+          # - playwright.availability.config.ts # TODO: alternative config file name.  Possibly for different reporters.
+      test-filter:
+        description: Test Filter - input the test name to run.
+        required: false
+        default: '' # TODO: Add a default test filter if needed.
+
+permissions:
+  id-token: write
+  contents: read
+  actions: read
+
+env:
+  DISPATCH_REF: ${{ inputs.tag == 0 && github.ref_name || inputs.branch-tag-sha }}
+  PLAYWRIGHT_TEST_DIR: 'tests/playwright' # TODO: Update this to the path of the playwright test root door
+  PLAYWRIGHT_REPORT_DIR: 'tests/playwright/playwright-report' # TODO: Update this to the path of the playwright report directory
+  PLAYWRIGHT_RESULTS_DIR: 'tests/playwright/test-results' # TODO: Update this to the path of the playwright test results directory
+  TIMEZONE: America/Denver
+  NPM_CACHE_PATH: '**/tests/playwright/node_modules' # TODO: Update this to the path of the node_modules directory in the playwright test root directory
```

Is there a reason this start of this path is different from the other path variables above?
```suggestion
  NPM_CACHE_PATH: 'tests/playwright/node_modules' # TODO: Update this to the path of the node_modules directory in the playwright test root directory
```

---

This ensures that it captures any node_modules.  I have this if the pathing is incorrect.  I have seen problems where the cache isn't saved when I specify the path and the working directory isn't what I expect.  I will be keeping this as it doesn't hurt to have it.  I feel it is a best practice.